listenbrainz/db/dump.py

```diff
@@ -301,6 +349,96 @@ def create_public_dump(location, dump_time, threads=DUMP_DEFAULT_THREAD_COUNT):
     )
 
 
+def create_feedback_dump(location, dump_time, threads=DUMP_DEFAULT_THREAD_COUNT):
+    """ Create a spark format dump of user listen and user recommendation feedback.
+    """
+    return _create_dump(
+        location=location,
+        dump_type='feedback',
+        tables=[],
+        dump_time=dump_time,
+        threads=threads,
+    )
+
+
+def dump_user_feedback(connection, location):
+    """ Carry out the actual dumping of user listen and user recommendation feedback.
+    """
+
+    with connection.begin() as transaction:
+
+        # First dump the user feedback
+        result = connection.execute(sqlalchemy.text("""
+            SELECT musicbrainz_id, recording_msid, score, r.created,
+                   EXTRACT(YEAR FROM r.created) AS year,
+                   EXTRACT(MONTH FROM r.created) AS month,
+                   EXTRACT(DAY FROM r.created) AS day
+              FROM recording_feedback r
+              JOIN "user"
+                ON r.user_id = "user".id
+          ORDER BY created"""))
+
+        last_day = ()
+        todays_items = []
+
+        while True:
+            row = result.fetchone()
+            today = (row[4], row[5], row[6]) if row else ()
```

Comment:
sqlalchemy should turn a date field in postgres into a python datetime (or date) here, allowing us to use `thedate.year` in order to understand what this is doing more clearly. Are you doing this to avoid the conversion on the assumption that it'll be faster?

If it's significantly faster and you'd prefer to keep it like this, please use dictionary access into the row: `row['month']` in order to make it clearer what rows you're using and to avoid bugs if we update the select statement